modules/Slider.js

```diff
@@ -0,0 +1,214 @@
+    o.c.rounded = o.c.rounded?20:0;
+
+    o.c.STEP_SIZE = ((!o.c.horizontal?o.c._height:o.c._width)-(!o.c.rounded?0:(2*o.c.rounded-7)))/o.c.steps;
+
+    // Add a rectangle object with x, y, x2, y2, w and h values.
+    o.c.r = {x:o.c.xStart, y:o.c.yStart, x2:o.c.xStart+o.c.width, y2:o.c.yStart+o.c.height, w:o.c.width, h:o.c.height};
+
+    // Initialize the level
+    o.v.level = (o.c.currLevel||o.c.currLevel===0)?o.c.currLevel:o.c.steps/2;
+
+    // Only add interactivity if wanted.
+    if (o.c.dragableSlider) { 
+
+      o.v.ebLast = 0;
+      o.v.dy = 0;
+
+      o.f.wasOnDragRect = (exFirst, eyFirst)=>{
+        "ram";
+        return exFirst>o.c.dragRect.x && exFirst<o.c.dragRect.x2 && eyFirst>o.c.dragRect.y && eyFirst<o.c.dragRect.y2;
+      };
+
+      o.f.wasOnIndicator = (exFirst)=>{
+        "ram";
+        if (!o.c.horizontal) return exFirst>o.c._xStart-o.c.oversizeL*o.c._width && exFirst<o.c._xStart+o.c._width+o.c.oversizeR*o.c._width;
+        if (o.c.horizontal) return exFirst>o.c._yStart-o.c.oversizeL*o.c._height && exFirst<o.c._yStart+o.c._height+o.c.oversizeR*o.c._height;
+      };
+
+      // Function to pass to `Bangle.on('drag', )`
+      o.f.dragSlider = e=>{
+        "ram";
+        if (o.v.ebLast==0) {
+          exFirst = o.c.horizontal?e.y:e.x;
+          eyFirst = o.c.horizontal?e.x:e.y;
+        }
+
+        // Only react if on allowed area.
+        if (o.f.wasOnDragRect(exFirst, eyFirst)) {
+          o.v.dragActive = true;
+          if (!o.c.propagateDrag) E.stopEventPropagation&&E.stopEventPropagation();
+
+          if (o.v.timeoutID) {clearTimeout(o.v.timeoutID); o.v.timeoutID = undefined;}
+          if (e.b==0 && !o.v.timeoutID && (o.c.timeout || o.c.timeout===0)) o.v.timeoutID = setTimeout(o.f.remove, 1000*o.c.timeout);
+
+          if (o.c.useMap && o.f.wasOnIndicator(exFirst)) { // If draging starts on the indicator, adjust one-to-one.
+
+            let input = !o.c.horizontal?
+              Math.min((175-e.y)-o.c.yStart-3*o.c.rounded/4, o.c.height):
+              Math.min(e.x-o.c.xStart-3*o.c.rounded/4, o.c.width);
+            input = Math.round(input/o.c.STEP_SIZE);
+
+            o.v.level = Math.min(Math.max(input,0),o.c.steps);
+
+            if (o.v.level != o.v.prevLevel) {
+              cb("map",o.v.level);
+              o.f.draw&&o.f.draw(o.v.level);
+            }
+            o.v.prevLevel = o.v.level;
+          } else if (o.c.useIncr) { // Heavily inspired by "updown" mode of setUI.
+
+            o.v.dy += o.c.horizontal?-e.dx:e.dy;
+            //if (!e.b) o.v.dy=0;
+
+            while (Math.abs(o.v.dy)>32) {
+              let incr;
+              if (o.v.dy>0) { o.v.dy-=32; incr = 1;}
+              else { o.v.dy+=32; incr = -1;}
+              Bangle.buzz(20);
+
+              o.v.level = Math.min(Math.max(o.v.level-incr,0),o.c.steps);
+              cb("incr",incr);
+              o.f.draw&&o.f.draw(o.v.level);
+              o.v.prevLevel = o.v.level;
```

Comment:
> (a bit like what you did with the border drawing / initiating it on first launch)

I don't think I do this anymore, right? Or can you point me to what you mean?

> Would the user notice if we moved the drawing to outside this while loop?

I think it would render a different behavior. But I'm not certain. I can test it. :+1:

Reply:
Tested it. It works well moving out drawing and updating `prevLevel` outside of the if-statement.

Reply:
I kind of like that the callback reacts on every increment though. That would change with your suggestion here I think?

Reply:
That's a good point, it would change and if it's snappy enough already with each increment let's keep it as is :)